README.md

```diff
@@ -33,309 +33,181 @@ Using RealtimeSanitizer requires RADSan-enabled `clang` and two actions:
-
-Building the Docker image locally is straightforward:
-
-```sh
-make docker
-```
-
-## Running the tests
-
-RADSan follows the existing sanitizer testing patterns, and adds two new test
-targets to the compiler-rt project for each architecture `arch`:
+## Error modes
 
-1. `TRadsan-${arch}-Test` (which is instrumented by RADSan), and
-2. `TRadsan-${arch}-NoInstTest` (which are unit tests that do not need the RADSan instrumentation)
-3. The "lit tests" which compile and run tests programs from scratch (see llvm-project/compiler-rt/test/radsan/TestCases/)
+At this time, RealtimeSanitizer has only a single mode of operation when a
+real-time violation is detected: the program will print an error message and
+terminate. More error modes are being integrated into the upstream project, and
+they will be documented here when they've been released.
 
-To run the tests:
+# Using RTSan with other compilers
 
-```sh
-make test
-```
-
-To run the full test suite of llvm compiler-rt (very slow, may have additional unexpected failures):
-
-```sh
-make check-compiler-rt
-```
+Yes, it's possible. Here's how.
 
-# Roadmap
+(TODO)
 
-The RADSan project keeps a public backlog and roadmap on
-[Trello](https://trello.com/b/6wpFlkqU). This board is used to distill and
-prioritise tasks originating from any source, including GitHub issues and pull
-requests, which we encourage anyone to submit.
+# Upstream Integration Roadmap
 
+- [x] Minimum viable functionality: raise an error if a blocking system library function is called in a real-time context (e.g. `malloc`)
+- [x] Ability to disable error reporting in a specified scope (`__rtsan::ScopedDisabler`)
+- [ ] Raise an error if a function attributed with `[[clang::blocking]]` is called in a real-time context
+  - Integration upstream is [in progress](https://github.com/llvm/llvm-project/pull/106754)
+- [ ] `Continue` error mode
+  - Integration upstream is [in progress](https://github.com/llvm/llvm-project/pull/107174)
+- [ ] Suppressing specific errors
 
 # Contact
 
-RealtimeSanitizer was originally developed by David Trevelyan and Ali Barker.
-We are grateful for the valuable contributions from Chris Apple, and
-welcome further contributions to make this tool more helpful to a wider group
-of developers. For all comments, suggestions and queries, please contact us by:
+RealtimeSanitizer was invented by David Trevelyan and Ali Barker, and the
+upstream integration was authored by Chris Apple and David Trevelyan, who
+continue its maintenance. We welcome further contributions to make this tool
+more helpful to a wider group of developers. For all comments, suggestions and
+queries, please contact us by:
 
-1. joining the [Discord server](https://discord.gg/DZqjbmSZzZ), or
-2. sending an email to realtime.sanitizer@gmail.com.
+1. joining the [Discord server](https://discord.gg/DZqjbmSZzZ),
+2. sending an email to realtime.sanitizer@gmail.com, or
+3. raising an issue in this GitHub repository.
```

Comment:
Do we still want to accept issues in this repo? Or should be say "file against the LLVM repo and tag us"?

Reply:
Good question. I think there's merit in allowing people to raise issues in either. If it's a bug report, the LLVM repo makes most sense. If it's a "hey, what do you RTSan guys think about this feature idea?", then it'd be easiest to track here, probably - saves it getting buried amongst all the other LLVM stuff. What do you think?

Reply:
That seems good to me, I guess it will likely be an evolving scenario, we can see what kind of issues/bugs we get and if we need to refine the approach over time.